### PR TITLE
adding appsync trigger-by

### PR DESCRIPTION
**Pull request by GuyMoses**

### CODE REVIEW

File: src/lumigo_tracer/event/event_trigger.py [hunk 1]
```diff
@@ -133,10 +135,22 @@ def _is_event_bridge(event: dict):
     )
 
 
+def _is_appsync(event: dict) -> bool:
+    host = event.get("context", {}).get("request", {}).get("headers", {}).get("host")
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:139]: `safe_get(event, ["context", "request", "headers", "host"])`
(your code will break on `event = {"context": "a"}`)

File: src/lumigo_tracer/event/event_trigger.py [hunk 2]
```diff
@@ -133,10 +135,22 @@ def _is_event_bridge(event: dict):
...
+    return host and "appsync-api" in host
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:140]: `and isinstance(host, str)`
(your code will break on `host=5`)

File: src/lumigo_tracer/event/event_trigger.py [hunk 3]
```diff
@@ -133,10 +135,22 @@ def _is_event_bridge(event: dict):
...
+
+
 def _parse_event_bridge(event: dict):
     return {"triggeredBy": "eventBridge", "messageId": event["id"]}
 
 
+def _parse_appsync(event: dict) -> dict:
+    headers = event.get("context", {}).get("request", {}).get("headers", {})
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:148]: same as before

File: src/lumigo_tracer/event/event_trigger.py (hunk 2 above)
Comment (saartochner) [reply src/lumigo_tracer/event/event_trigger.py:140]: `and '.' in host`

File: src/lumigo_tracer/event/event_trigger.py [hunk 4]
```diff
@@ -133,10 +135,22 @@ def _is_event_bridge(event: dict):
...
+    api_id = headers.get("host", "").split(".")[0]
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:149]: @doriaviram wdyt? Should we prefer to do this parsing in the BE?

File: src/lumigo_tracer/event/event_trigger.py [hunk 5]
```diff
@@ -133,10 +135,25 @@ def _is_event_bridge(event: dict):
...
+
+
 def _parse_event_bridge(event: dict):
     return {"triggeredBy": "eventBridge", "messageId": event["id"]}
 
 
+def _parse_appsync(event: dict) -> dict:
+    headers = safe_get(event, ["context", "request", "headers"])
+    host = headers.get("host")
+    api_id = host.split(".")[0] if host and isinstance(host, str) else None
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:150]:
```suggestion
    api_id = host.split(".")[0]
```
You can assume it :)

Comment (GuyMoses) [reply src/lumigo_tracer/event/event_trigger.py:150]: can I assume it's a string?

Comment (saartochner) [reply src/lumigo_tracer/event/event_trigger.py:150]: I think - you checked it in `_is_appsync`...

Comment (GuyMoses) [reply src/lumigo_tracer/event/event_trigger.py]: oh yeah, but code pieces could change for some reason

Comment (saartochner) [reply src/lumigo_tracer/event/event_trigger.py]: 👍

**Review by saartochner — APPROVED** [review]
Well done!
Let's just wait for Dori's suggestion on the tracer-BE separation.
The code is approved on my side

**Pull request closed without merging**